### PR TITLE
refactor/contextAPI-implementation

### PR DESCRIPTION
**Pull request by GeorgeLR87**

### CODE REVIEW

File: src/components/Cards/components/BackCard.tsx [hunk 1]
```diff
@@ -1,13 +1,17 @@
...
+const BackCard = () => {
 
-const BackCard = ({cvc}: BackCardProps) => (
-    <div className="bg-[url('assets/images/bg-card-back.png')] bg-cover w-72 h-40 md:w-96 md:h-52 rounded-lg p-6 text-white text-xs md:text-sm tracking-widest lg:col-start-2 lg:col-end-5 lg:static absolute top-5 right-5 sm:top-9 sm:right-6">
-        <p className="h-full flex justify-end items-center pr-2.5 md:pr-5">
-            { cvc|| '000'}
-        </p>
-    </div>
-)
+    const { state } = useForm()
```
Comment (DaroMacs) [inline src/components/Cards/components/BackCard.tsx]: in my opinion this custom hook is not needed that's why you created the context. You could make this directly with the context instead calling the useForm

```
const BackCard = () => {
  const { state } = useContext(FormContext);
  const { cvc } = state;

```

File: src/components/Cards/components/FrontCard.tsx [hunk 2]
```diff
@@ -1,34 +1,31 @@
...
-                <p className="w-full text-lg md:text-2xl text-left tracking-widest">
-                    {cardNumber || '0000 0000 0000 0000'}
-                </p>
-                <div className="flex justify-between text-xs md:text-sm uppercase tracking-widest">
-                    <p>{cardholderName || 'Jane Appleseed'}</p>
-                    <p>{expMonth || '00'}/{expYear || '00'}</p>
+
+const FrontCard = () => {
+
+    const { state } = useForm()
```
Comment (DaroMacs) [inline src/components/Cards/components/FrontCard.tsx]: Same here with the context

```
const FrontCard = () => {
  const { state } = useContext(FormContext);
  const { cardNumber, cardholderName, expMonth, expYear } = state;
```

File: src/components/Dates/components/ExpMonthInput.tsx [hunk 3]
```diff
@@ -1,16 +1,15 @@
...
-type ExpMonthInputProps = {
-    state: string
-    dispatch: Dispatch<FormActions>
-    expMonthError?: string;
-    expYearError?: string;
-}
+const ExpMonthInput = () => {
 
-const ExpMonthInput = ({ state, dispatch, expMonthError, expYearError }: ExpMonthInputProps) => {
+    const { state, dispatch, errors } = useForm()
```
Comment (DaroMacs) [inline src/components/Dates/components/ExpMonthInput.tsx]: Same here replace the hook directly with the context

File: src/components/Dates/components/ExpYearInput.tsx [hunk 4]
```diff
@@ -1,15 +1,14 @@
...
+import { useForm } from "../../../hooks/useForm"
 
-type ExpYearInputProps = {
-    state: string;
-    dispatch: Dispatch<FormActions>;
-    error?: string;
-}
+const ExpYearInput = () => {
+
+    const { state, dispatch, errors } = useForm()
```
Comment (DaroMacs) [inline src/components/Dates/components/ExpYearInput.tsx]: Same here replace the hook directly with the context

File: src/components/Dates/components/ExpYearInput.tsx [hunk 5]
```diff
@@ -1,15 +1,14 @@
...
+    const { expYear } = state
+    const expYearError = errors[EFormData.ExpYear];
 
-const ExpYearInput = ({ state, dispatch, error }: ExpYearInputProps) => {
     const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
```
Comment (DaroMacs) [inline src/components/Dates/components/ExpYearInput.tsx]: this handleChange should be ideal to have it in the context so you remove most of the logic to the context and keep clean the component

File: src/components/Form/Form.tsx [hunk 6]
```diff
@@ -1,21 +1,13 @@
...
-  dispatch: Dispatch<FormActions>;
-  setIsSubmitted: Dispatch<SetStateAction<boolean>>;
-};
+const Form = () => {
 
-const Form = ({ state, dispatch, setIsSubmitted }: FormProps) => {
-  const [errors, setErrors] = useState<FormErrors>({});
+  const { state, setErrors, setIsSubmitted } = useForm()
 
   const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
```
Comment (DaroMacs) [inline src/components/Form/Form.tsx]: this handleChange should be ideal to have it in the context so you remove most of the logic to the context and keep clean the component

File: src/components/CardSuccess.tsx [hunk 7]
```diff
@@ -1,13 +1,10 @@
...
-type FormProps = {
-  dispatch: Dispatch<FormActions>;
-  setIsSubmitted: Dispatch<SetStateAction<boolean>>;
-};
+const CardSuccess = () => {
+
+  const { dispatch, setIsSubmitted } = useForm()
 
-const CardSuccess = ({ dispatch, setIsSubmitted }: FormProps) => {
   const handleClick = () => {
```
Comment (DaroMacs) [inline src/components/CardSuccess.tsx]: this handleChange should be ideal to have it in the context so you remove most of the logic to the context and keep clean the component

File: src/components/Dates/components/ExpMonthInput.tsx [hunk 8]
```diff
@@ -1,16 +1,15 @@
...
+    const { expMonth } = state
+    const expMonthError = errors[EFormData.ExpMonth];
+    const expYearError = errors[EFormData.ExpYear];
+    
 
     const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
```
Comment (DaroMacs) [inline src/components/Dates/components/ExpMonthInput.tsx]: this handleChange should be ideal to have it in the context so you remove most of the logic to the context and keep clean the component

File: src/components/Form/components/CVCInput.tsx [hunk 9]
```diff
@@ -1,15 +1,14 @@
...
-  error?: string;
-};
+const CVCInput = () => {
+
+  const { state, dispatch, errors } = useForm()
+  const { cvc } = state
+  const cvcError = errors[EFormData.CVC];
 
-const CVCInput = ({ state, dispatch, error }: CVCInputProps) => {
   const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
```
Comment (DaroMacs) [inline src/components/Form/components/CVCInput.tsx]: this handleChange should be ideal to have it in the context so you remove most of the logic to the context and keep clean the component

File: src/components/Form/components/CardHolderNameInput.tsx [hunk 10]
```diff
@@ -1,14 +1,13 @@
...
-  error?: string;
-};
+const CardHolderNameInput = () => {
+
+  const { dispatch, state, errors } = useForm()
+  const {cardholderName} = state
+  const cardholderNameError = errors[EFormData.CardholderName];
 
-const CardHolderNameInput = ({ state, dispatch, error }: CardholderNameInputProps) => {
   const handleChange = (
```
Comment (DaroMacs) [inline src/components/Form/components/CardHolderNameInput.tsx]: this handleChange should be ideal to have it in the context so you remove most of the logic to the context and keep clean the component

File: src/hooks/useForm.ts [hunk 11]
```diff
@@ -0,0 +1,11 @@
+import { useContext } from "react"
```
Comment (DaroMacs) [inline src/hooks/useForm.ts]: Remove this hook

**Review by dario-macedo-nftStudios — APPROVED** [review]

**Pull request merged**